src/main/java/assets/game.js

```diff
@@ -261,15 +280,19 @@ function initGame() {
 
     document.getElementById("place_minesweeper").addEventListener("click", function(e) {
         shipType = "MINESWEEPER";
-       registerCellListener(place(2));
+       registerCellListener(place(2, false));
     });
     document.getElementById("place_destroyer").addEventListener("click", function(e) {
         shipType = "DESTROYER";
-       registerCellListener(place(3));
+       registerCellListener(place(3, false));
     });
     document.getElementById("place_battleship").addEventListener("click", function(e) {
         shipType = "BATTLESHIP";
-       registerCellListener(place(4));
+       registerCellListener(place(4, false));
+    });
+    document.getElementById("place_submarine").addEventListener("click", function(e) {
+            shipType = "SUBMARINE";
```

Comment:
This works well using the ship types because it makes it easier to add even more ships that are the same type. Good object oriented design!